lib/rules/no-unassigned-vars.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @fileoverview Rule to flag variables that are never assigned
+ * @author Jacob Bandes-Storch <https://github.com/jtbandes>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../types').Rule.RuleModule} */
+module.exports = {
+	meta: {
+		type: "problem",
+		dialects: ["typescript", "javascript"],
+		language: "javascript",
+
+		docs: {
+			description:
+				"Disallow variable declarations that use `let` or `var` but have no initial value and are never assigned",
```

Comment:
`no-unused-vars` will also [report vars](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBcImVycm9yXCIgKi9cblxubGV0IHg7XG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=) with this description.

So something like this is more accurate:

```suggestion
			description:
				"Disallow `let` or `var` variables that are read but never assigned.",
```

Reply:
Doesn't look like there is a trailing `.` on most rules' descriptions

Reply:
Yeah, we can remove that. Added accidently